### DISCUSSION TITLE
How to see a list of all open (sub) databases?

### DISCUSSION THREAD

**Discussion by azarenkovgd** [kriszyp/lmdb-js#215] [Q&A] [answered]
Hi!

How to see a list of all open (sub) databases for <dbname> (opened with <dbname>.opendb...)?
I searched the documentation and the api but didn't find anything that fits, maybe I missed something.
Of course, you can keep a list of all open databases yourself, but it does not look as elegant as the native solution and is prone to errors.
Scenario - cleaning up unused (sub) databases.

Thank you!

**Comment by kriszyp** [comment, marked as answer]
You can do a getRange() on the root database (where name is undefined or null), to get a list of all the databases, as they are all stored there. The keys for the range will be the database names. However, you need to make sure your root database has an `encoding: 'binary'` or you may get decoding errors.

**Reply by azarenkovgd** [reply]
Thanks!
